### DISCUSSION TITLE
[Feature] It is recommended to separate http and https in nginx site configuration, which is more in line with the standard.

### DISCUSSION THREAD

**Discussion by gegj** [1Panel-dev/1Panel#7526] [Ideas]
### 1Panel Version

v1.10.22-lts

### Please describe your needs or suggestions for improvements

建议nginx配置把http和https分开来，比如添加hsts的时候，就只需要添加到https上就行，http是不需要加的。
https://www.ssl.org/server-config-generator

server {
    listen 80 default_server;
    listen [::]:80 default_server;

    location / {
        return 301 https://$host$request_uri;
    }
}

server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;

    ssl_certificate /path/to/signed_cert_plus_intermediates;
    ssl_certificate_key /path/to/private_key;
    ssl_session_timeout 1d;
    ssl_session_cache shared:MozSSL:10m;  # about 40000 sessions
    ssl_session_tickets off;
![PixPin_2024-12-22_12-12-17](https://github.com/user-attachments/assets/d37abbd7-4d37-40c0-8aea-22d36e324f2f)

**Comment by gegj** [comment]
![PixPin_2024-12-22_12-16-33](https://github.com/user-attachments/assets/1479bdfd-5c40-4d83-805d-10ddc0223444)
添加https://hstspreload.org/ 的时候的提示

**Comment by zhengkunwang223** [comment]
如果分开配置，那么所有除 ssl 的配置都要写两份，从逻辑上来讲，是不好处理的。如果有特殊需要，建议还是自行修改配置

**Comment by wanghe-fit2cloud** [comment]
> Bot detected the issue body's language is not English, translate it automatically. 👯👭🏻🧑‍🤝‍🧑👫🧑🏿‍🤝‍🧑🏻👩🏾‍🤝‍👨🏿👬🏿         
----    
If configured separately, all configurations except ssl must be written in two copies, which is logically difficult to handle. If you have special needs, it is recommended to modify the configuration yourself.

**Comment by wanghe-fit2cloud** [comment]
将 Issue 转为 Discussion.
